packages/sanity/src/core/tasks/components/form/tasksFormBuilder/FormCreate.tsx

```diff
@@ -47,7 +44,52 @@ export function FormCreate(props: ObjectInputProps) {
   }, [setViewMode])
   const {handleRemove, removeStatus} = useRemoveTask({id: value._id, onRemoved: onRemove})
   const {t} = useTranslation(tasksLocaleNamespace)
-  const handleCreate = useCallback(() => {
+  const {data} = useTasks()
+  const savedTask = data.find((task) => task._id === value._id)
+
+  useEffect(() => {
+    // This useEffect takes care of closing the form when a task entered the "creation" state.
+    // That action is async and we don't have access to the promise, once the value is updated in the form we will close the form.
+    if (creating && savedTask?.createdByUser) {
+      telemetry.log(TaskCreated)
+      toast.push({
+        closable: true,
+        status: 'success',
+        title: t('form.status.success'),
+      })
+
+      setCreating(false)
+      if (createMore) {
+        setViewMode({type: 'create'})
+      } else {
+        setActiveTab('subscribed')
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [creating, savedTask?.createdByUser])
+
+  useEffect(() => {
+    // If after 10 seconds the task is still in the "creating" state, show an error and reset the creating state.
+    let timeoutId: NodeJS.Timeout | null = null
```

Comment:
We should avoid references to NodeJS types since this targets a browser runtime

```suggestion
let timeoutId: ReturnType<typeof setTimeout> | null = null
```